extensions/nv/GLSL_NV_shader_invocation_reorder.txt

```diff
@@ -0,0 +1,848 @@
+
+    Returns implementation defined index of geometry encoded in the hit
+    object.
+
+
+    Syntax:
+
+        int hitObjectGetPrimitiveIndexNV(hitObjectNV hitObject);
+
+    Returns the index of the primitive (triangle or bounding box) within the
+    geometry as encoded in the hit object.
+
+
+    Syntax:
+
+        uint hitObjectGetHitKindNV(hitObjectNV hitObject);
+
+    Returns values as defined in EXT_ray_tracing specification for
+    gl_HitKindEXT.
+
+
+    Syntax:
+
+        void hitObjectGetAttributesNV(hitObjectNV hitObject, int attributeLocation);
+
+    
+    Extracts the attributes encoded in the hit object and writes to 
+    'hitObjectAttributeNV' storage class decorated variable selected using
+    'attributeLocation' as specified below.
+
+    <attributeLocation> is a compile-time constant to select
+    a shader defined structure used to populate with attributes associated
+    with this hit object. 
+    It is possible for a shader to contain multiple invocations
+    'hitObjectGetAttributesNV'
+    with different attribute types. Different attribute types are chosen based on the 
+    different values of the compile-time constant <attributeLocation> which correspond 
+    to the hitObjectAttributeNV qualified variables having the same value for the
+    location layout qualifier. 
+
+    
+    Syntax:
+
+        uvec2 hitObjectGetShaderRecordBufferHandleNV(hitObjectNV hitObject);
+
+    
+    Returns a uvec2 handle representing the address for data for a record in
+    the shader binding table as encoded in the hit object.
+    
+    This can be used with a buffer reference with additional layout qualifier 
+    'hitobjectshaderrecordnv' as defined in EXT_buffer_reference_uvec2.
+
+
+    Syntax:
+
+        uint hitObjectGetShaderBindingTableRecordIndexNV(hitObjectNV hitObject);
+
+    Returns the index for the record in shader binding table as encoded in hit
+    object. For a hit, returns index within hit group for a miss the index within
+    miss table.
+
+
+    Syntax:
+    
+        float hitObjectGetCurrentTimeNV(hitObjectNV hitObject);
+
+    Returns the current time as encoded in the hit object.
+    Only applicable if hit object was constructed for hitObjectTraceRayMotionNV
+
+
+    Add Section 8.20, Reorder Functions
+
+
+    Reorder functions allow implementations to reorder threads based
+    on input given to these builtins to so that subsequent code can
+    execute with improved coherence.
+
+    For reorder functions providing a hint, the implementation will reorder 
+    based on the unsigned numerical value of the number of bits provided. 
+    The sorting may be coarse, be bucketing rather than sorting, or only use 
+    the high bits of the hint provided. If both the hint and the hit object 
+    are provided the implementation should reorder based on the hint after 
+    the shader being executed. The value of bits of a hint should be <= 32
```

Comment:
The number of bits must be <= 32 (how could it be more?) Or the value of the hint must be <= 32, assuming the later...

```suggestion
    the shader being executed. The value of a hint should be <= 32.
```

Reply:
Oh I understand now after reading further, but this seems redundant with the statements for each of the instructions below. Maybe just drop these last two sentences?